src/Symfony/Bundle/FrameworkBundle/Routing/Router.php

```diff
@@ -51,7 +51,11 @@ public function __construct(ContainerInterface $container, $resource, array $opt
     public function getRouteCollection()
     {
         if (null === $this->collection) {
+            $this->collection = new RouteCollection();
```

Comment:
The thing is that the getter changes the state of the property. And this state transition is the responsibility of the Router class, not of its caller (which does not know about this).
An other fact is that the loader can fail with a fatal error, which means that the caller will be bypassed.
But the fatal error being caught will be re-injected for display by the Debug component, and this code patch will be taken a second path. Only this code can know about that.